src/openrct2/scenario/ScenarioSources.h

```diff
@@ -18,6 +18,7 @@ struct SourceDescriptor
     uint8_t source;
     int32_t index;
     uint8_t category;
+    const char* textObjectId;
```

Comment:
Should this be a C-style string?

Reply:
I think so, at least for now. It currently requires less `.c_str()` hassle down the line.

Reply:
I’m sure if I am a fan of that. Legacy stuff has a tendency to stick around, better not to introduce any more of it.

Reply:
Sometimes basic data types are the better choice, though. I wouldn't mind calling it `const utf8*`, if that feels better?

Reply:
That’s barely an improvement. Why are you so hell-bent on using a C string anyway? Am I missing something?

Reply:
Why not a std::string_view?

Reply:
That’s fine by me. I’m not bothered whether it is `u8string` or `u8string_view`.

Reply:
I've attempted this again, but if I use `u8string_view`, it can't unify with `AppendRequiredObjects` (expects std::string compatible vector). Conversely, if I try `u8string` or `std::string`, the constexpr initialisation of `ScenarioTitleDescriptor` fails.

I really would like to leave this as-is.

Reply:
Reworking `AppendRequiredObjects` to work with views instead solves the issue.